### PR TITLE
Update Nextclade metadata merge to use `augur curate rename` and `augur merge`

### PR DESCRIPTION
**Pull request by tsibley**
See individual commit messages for all the details.

Marking this PR as draft (for now) because it's primarily meant (for now) to solicit feedback on this approach. Once we're happy with the implementation and come to a consensus, we can merge it and then port it to [nextstrain/pathogen-repo-guide](https://github.com/nextstrain/pathogen-repo-guide).

## Checklist

- [x] Checks pass

### CODE REVIEW

**Comment by tsibley** [comment]
To run this with the unreleased [`--no-source-columns` option to `augur merge`](https://github.com/nextstrain/augur/pull/1625), use the Docker runtime with a local Augur overlay, e.g.

    nextstrain build --docker --augur ../augur/ ingest -f results/metadata.tsv

or run in an ambient runtime with [Augur installed from source](https://docs.nextstrain.org/projects/augur/en/stable/installation/installation.html)

    nextstrain build --ambient ingest -f results/metadata.tsv

**Comment by tsibley** [comment]
CI is failing because it requires an unreleased Augur feature at the moment.

File: ingest/rules/nextclade.smk [hunk 1]
```diff
@@ -46,28 +48,33 @@ rule run_nextclade:
...
-            -f '*' \
-            -p '(.+)' \
-            -r '{{kv}}' \
-            -k {input.nextclade_field_map} \
+        augur curate rename \
+            --metadata {input.nextclade:q} \
+            --id-column {params.nextclade_id_field:q} \
+            --field-map {params.nextclade_field_map:q} \
+            --output-metadata - \
+        | tsv-select --header --fields {params.nextclade_fields:q} \
```
Comment (joverlee521) [inline ingest/rules/nextclade.smk:78]: _tangent_

I was thinking of revisiting https://github.com/nextstrain/augur/issues/1526 to remove the need to use `tsv-select` here, but now `tsv-select` looks so much cleaner than having to list out the columns to drop 😅

**Review by joverlee521 — APPROVED** [review]
Thanks for using measles as an example! This is so much cleaner 🌟

File: ingest/rules/nextclade.smk [hunk 2]
```diff
@@ -62,6 +62,7 @@ rule join_metadata_and_nextclade:
         metadata="data/subset_metadata.tsv",
     output:
         metadata="results/metadata.tsv",
+        nextclade_metadata=temp("results/nextclade_metadata.csv"),
```
[the remark this answers is not in the document]
Comment (tsibley) [reply ingest/rules/nextclade.smk:65]: Ahh, good to know, thanks! I can split it out.

[the remark this answers is not in the document]
Comment (tsibley) [reply ingest/rules/nextclade.smk]: Fixed in repush.

File: ingest/rules/nextclade.smk (hunk 1 above)
Comment (tsibley) [reply ingest/rules/nextclade.smk:75]: `tsv-select` will be less clean if we were properly using `csv2tsv --csv-delim $'\t' | tsv-select …`.

Comment (joverlee521) [reply ingest/rules/nextclade.smk:75]: Oh, hmm. Depending on the type of TSV that Nextclade outputs, we could flip it to `tsv-select ... | augur curate rename`?

Comment (tsibley) [reply ingest/rules/nextclade.smk:75]: [Nextclade outputs CSV-like TSV](https://github.com/nextstrain/augur/issues/1566#issuecomment-2344625695), so we will still need the `csv2tsv …` to use tsv-utils.

Comment (tsibley) [reply ingest/rules/nextclade.smk:75]: I'm going to leave the precise TSV-flavors and correct handling of them as future work for https://github.com/nextstrain/augur/issues/1566. Lots of tsv-utils and csvtk usages will have to change if we want them to be correct (and I think we do).

File: ingest/rules/nextclade.smk [hunk 3]
```diff
@@ -46,35 +48,53 @@ rule run_nextclade:
...
+    with open(config["nextclade"]["field_map"], "r") as f:
+        config["nextclade"]["field_map"] = dict(line.rstrip("\n").split("\t", 1) for line in f if not line.startswith("#"))
+
+
+rule nextclade_metadata:
     input:
         nextclade="results/nextclade.tsv",
+    output:
+        nextclade_metadata=temp("results/nextclade_metadata.tsv"),
+    params:
```
Comment (genehack) [inline ingest/rules/nextclade.smk:64]: Missing `log` and `benchmark`, which I think we're trying to include everywhere?

**Review by j23414 — APPROVED** [review]
Nice!

File: ingest/rules/nextclade.smk [hunk 4]
```diff
@@ -46,35 +48,53 @@ rule run_nextclade:
...
+
+    with open(config["nextclade"]["field_map"], "r") as f:
+        config["nextclade"]["field_map"] = dict(line.rstrip("\n").split("\t", 1) for line in f if not line.startswith("#"))
+
+
+rule nextclade_metadata:
     input:
         nextclade="results/nextclade.tsv",
+    output:
+        nextclade_metadata=temp("results/nextclade_metadata.tsv"),
```
Comment (genehack) [inline ingest/rules/nextclade.smk:63]: why `temp()` here?

File: ingest/rules/nextclade.smk [hunk 5]
```diff
@@ -46,35 +48,53 @@ rule run_nextclade:
...
+            --metadata {input.nextclade:q} \
+            --id-column {params.nextclade_id_field:q} \
+            --field-map {params.nextclade_field_map:q} \
+            --output-metadata - \
+        | tsv-select --header --fields {params.nextclade_fields:q} \
+        > {output.nextclade_metadata:q}
+        """
+
+
+rule join_metadata_and_nextclade:
```
Comment (genehack) [inline ingest/rules/nextclade.smk:80]: Also missing `log` and `benchmark` here.

**Review by genehack — APPROVED** [review]

File: ingest/rules/nextclade.smk (hunk 3 above)
Comment (tsibley) [reply ingest/rules/nextclade.smk:64]: These were missing in the rule prior to my changes. None of the rules in this file have those stanzas. I'm going to punt on changing that here and now.

File: ingest/rules/nextclade.smk (hunk 5 above)
Comment (tsibley) [reply ingest/rules/nextclade.smk:80]: [as above](https://github.com/nextstrain/measles/pull/52#discussion_r1761604605)

**Pull request merged**